lib/http.ts

```diff
@@ -36,15 +35,18 @@ export default class HTTPClient {
     return res.data as Readable;
   }
 
-  public postJson(url: string, body?: any): Promise<AxiosResponse> {
-    return this.instance.post(url, body, {
+  public async post<T>(url: string, body?: any): Promise<T> {
+    const res = await this.instance.post(url, body, {
       headers: { "Content-Type": "application/json" },
     });
-  }
+    let resBody = {
+      ...res.data,
+    };
+    if (res.headers["x-line-request-id"]) {
```

Comment:
I think that, It should not have Domain knowledges in the utility module like `http.ts`.
For example, When an instance of `HttpClient` is created, can you pass it as an optional parser function?